mutiny-core/src/lib.rs

```diff
@@ -2290,6 +2300,118 @@ impl<S: MutinyStorage> MutinyWallet<S> {
     pub fn is_safe_mode(&self) -> bool {
         self.safe_mode
     }
+
+    async fn cashu_mint_post<T: serde::de::DeserializeOwned>(
+        client: &reqwest::Client,
+        url: &Url,
+        body: Value,
+    ) -> Result<T, MutinyError> {
+        let res = client
+            .post(url.clone())
+            .header("Content-Type", "application/json")
+            .body(body.to_string())
+            .send()
+            .await
+            .map_err(|_| MutinyError::CashuMintError)?;
+        Self::parse_cashu_mint_response(res).await
+    }
+
+    async fn parse_cashu_mint_response<T: serde::de::DeserializeOwned>(
+        res: reqwest::Response,
+    ) -> Result<T, MutinyError> {
+        match res.status() {
+            StatusCode::OK => {
+                let response_text = res.text().await.map_err(|_| MutinyError::CashuMintError)?;
+                match serde_json::from_str::<T>(&response_text) {
+                    Ok(data) => Ok(data),
+                    Err(_) => Err(MutinyError::CashuMintError),
+                }
+            }
+            _ => {
+                let txt = res.text().await.map_err(|_| MutinyError::CashuMintError)?;
+                let data = serde_json::from_str::<CashuErrorResponse>(&txt)
+                    .map_err(|_| MutinyError::CashuMintError)?;
+
+                match data.code {
+                    // error code in nutshell for tokens that have been spent
+                    11001 => Err(MutinyError::TokenAlreadySpent),
+                    _ => Err(MutinyError::CashuMintError),
+                }
+            }
+        }
+    }
+
+    /// Calls upon a Cashu mint and redeems/melts the token.
+    pub async fn melt_cashu_token(
+        &self,
+        token_v3: TokenV3,
+    ) -> Result<Vec<MutinyInvoice>, MutinyError> {
+        let client = reqwest::Client::new();
```

Comment:
its kinda bad practice to init a new `reqwest` client for every request. It doesn't matter in the browser but for other build targets this will make us not do any of the nice connection pooling and stuff that reqwest would do for us.

if you make the `CashuHttpClient` that could have its own `reqwest::Client` so it doesn't to re-init every request.